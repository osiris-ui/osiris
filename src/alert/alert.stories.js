import { storiesOf } from '@storybook/vue';

import OAlert from './alert.vue';

storiesOf('Alert', module)
  // .add('default', () => '<o-row>simple row</o-row>');
  .add('types', () => ({
    components: { OAlert },
    template: `
      <div>
        <div style="margin-top: 10px">
          <o-alert type="primary">This is a primary alert</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="secondary">This is a secondary alert</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="info">This is an info alert</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="success">This is a success alert</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="warning">This is a warning alert</o-alert>
        </div>
      </div>
    `,
  }))
  .add('sizes', () => ({
    components: { OAlert },
    template: `
      <div>
        <div style="margin-top: 10px">
          <o-alert type="primary" size="large">Primary Large</o-alert>
          <o-alert type="primary">Primary Normal</o-alert>
          <o-alert type="primary" size="small">Primary Small</o-alert>
          <o-alert type="primary" size="mini">Primary Mini</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="secondary" size="large">Secondary Large</o-alert>
          <o-alert type="secondary">Secondary Normal</o-alert>
          <o-alert type="secondary" size="small">Secondary Small</o-alert>
          <o-alert type="secondary" size="mini">Secondary Mini</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="info" size="large">Info Large</o-alert>
          <o-alert type="info">Info Normal</o-alert>
          <o-alert type="info" size="small">Info Small</o-alert>
          <o-alert type="info" size="mini">Info Mini</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="success" size="large">Success Large</o-alert>
          <o-alert type="success">Success Normal</o-alert>
          <o-alert type="success" size="small">Success Small</o-alert>
          <o-alert type="success" size="mini">Success Mini</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="warning" size="large">Warning Large</o-alert>
          <o-alert type="warning">Warning Normal</o-alert>
          <o-alert type="warning" size="small">Warning Small</o-alert>
          <o-alert type="warning" size="mini">Warning Mini</o-alert>
        </div>
        <div style="margin-top: 10px">
          <o-alert type="danger" size="large">Danger Large</o-alert>
          <o-alert type="danger">Danger Normal</o-alert>
          <o-alert type="danger" size="small">Danger Small</o-alert>
          <o-alert type="danger" size="mini">Danger Mini</o-alert>
        </div>
      </div>
    `,
  }))
  .add('round', () => ({
    components: { OAlert },
    template: `
      <div>
        <o-alert type="primary" round>Primary</o-alert>
        <o-alert type="secondary" round>Secondary</o-alert>
        <o-alert type="info" round>Info</o-alert>
        <o-alert type="success" round>Success</o-alert>
        <o-alert type="warning" round>Warning</o-alert>
        <div style="margin-top:10px; margin-left: 0">
          <o-alert type="danger" round>Danger</o-alert>
        </div>
      </div>
    `,
  }));
