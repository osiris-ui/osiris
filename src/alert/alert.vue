<template>
  <div
    :class="[
      `o-alert--${type}`,
      `o-alert--${size}`,
      {
        'is-round': round,
      }
    ]"
    class="o-alert">
    <div>
      <div class="o-alert-icon">
        <i
          :class="[
            `o-icon-${icon}`
          ]"
          class="o-icon"/>
      </div>
      <div
        :class="[
          `o-alert-text--${type}`,
        ]"
        class="o-alert-text">
        <div>
          <slot />
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {

  name: 'OAlert',
  props: {
    icon: {
      type: String,
      default: 'check',
    },

    type: {
      type: String,
      default: 'primary',
    },

    size: {
      type: String,
      default: 'normal',
    },

    round: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
