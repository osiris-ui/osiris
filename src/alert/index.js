import alert from './alert.vue';

export default alert;
