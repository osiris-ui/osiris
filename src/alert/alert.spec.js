import { mount } from '@vue/test-utils';
import alert from './alert.vue';

describe('alert', () => {
  let wrapper;

  const reset = () => wrapper.setProps({
    type: 'primary',
    size: 'normal',
    round: false,
  });

  beforeAll(() => {
    wrapper = mount(alert);
  });

  afterEach(reset);

  it('should be a primary alert at startup', () => {
    const defaultClasses = ['o-alert', 'o-alert--primary', 'o-alert--normal'];

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toEqual(defaultClasses);
  });

  it('should be a primary alert', () => {
    wrapper.setProps({
      type: 'primary',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--primary');
  });

  it('should be a secondary alert', () => {
    wrapper.setProps({
      type: 'secondary',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--secondary');
  });

  it('should be a info alert', () => {
    wrapper.setProps({
      type: 'info',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--info');
  });

  it('should be a success alert', () => {
    wrapper.setProps({
      type: 'success',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--success');
  });

  it('should be a warning alert', () => {
    wrapper.setProps({
      type: 'warning',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--warning');
  });

  it('should be a danger alert', () => {
    wrapper.setProps({
      type: 'danger',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--danger');
  });

  it('should be a mini alert', () => {
    wrapper.setProps({
      size: 'mini',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--mini');
  });

  it('should be a small alert', () => {
    wrapper.setProps({
      size: 'small',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--small');
  });

  it('should be a normal alert', () => {
    wrapper.setProps({
      size: 'normal',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--normal');
  });

  it('should be a large alert', () => {
    wrapper.setProps({
      size: 'large',
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('o-alert--large');
  });

  it('should be a round alert', () => {
    wrapper.setProps({
      round: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('is-round');
  });
});
