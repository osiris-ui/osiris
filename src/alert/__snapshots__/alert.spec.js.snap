// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert should be a danger alert 1`] = `
<div class="o-alert o-alert--danger o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--danger">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a info alert 1`] = `
<div class="o-alert o-alert--info o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--info">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a large alert 1`] = `
<div class="o-alert o-alert--primary o-alert--large">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a mini alert 1`] = `
<div class="o-alert o-alert--primary o-alert--mini">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a normal alert 1`] = `
<div class="o-alert o-alert--primary o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a primary alert 1`] = `
<div class="o-alert o-alert--primary o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a primary alert at startup 1`] = `
<div class="o-alert o-alert--primary o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a round alert 1`] = `
<div class="o-alert o-alert--primary o-alert--normal is-round">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a secondary alert 1`] = `
<div class="o-alert o-alert--secondary o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--secondary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a small alert 1`] = `
<div class="o-alert o-alert--primary o-alert--small">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--primary">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a success alert 1`] = `
<div class="o-alert o-alert--success o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--success">
      <div></div>
    </div>
  </div>
</div>
`;

exports[`alert should be a warning alert 1`] = `
<div class="o-alert o-alert--warning o-alert--normal">
  <div>
    <div class="o-alert-icon">
      <i class="o-icon o-icon-check"></i>
    </div>
    <div class="o-alert-text o-alert-text--warning">
      <div></div>
    </div>
  </div>
</div>
`;
